rust/foxglove/src/websocket/server_listener.rs

```diff
@@ -61,4 +61,8 @@ pub trait ServerListener: Send + Sync {
     /// Callback invoked when the last client unsubscribes from the connection graph. Requires
     /// [`Capability::ConnectionGraph`][super::Capability::ConnectionGraph].
     fn on_connection_graph_unsubscribe(&self) {}
+    /// Callback invoked when a client connects to the server.
+    fn on_client_connect(&self) {}
+    /// Callback invoked when a client disconnects from the server.
+    fn on_client_disconnect(&self) {}
```

Comment:
We should also pass `Client` as one of the callback arguments, in case a callback implementer wants to know the client's ID or sink ID